Lib/asyncio/base_events.py

```diff
@@ -1794,7 +1794,16 @@ def call_exception_handler(self, context):
                              exc_info=True)
         else:
             try:
-                self._exception_handler(self, context)
+                ctx = None
+                task = context.get("task")
+                if task is None:
+                    task = context.get("future")
```

Comment:
Futures don't have context associated with them so `hasattr` will always return `False`. You can just remove it.

Reply:
But Task inherits from Future and when some Task method calls super().method() it will call a Future method which will create a context whose "future" key actually references a Task.